### PR TITLE
Fix for ZM8 - Kam'lanaut Fight CS issue #2756 :

### PR DESCRIPTION
**Pull request by dickeyf**
End BCNM 7 seconds after winning.
Play CS 6 (The final cutscene) after ZM8 BCNM is won.

Tested against Mission 5-2, ZM8 and a BCNM (Under observation).

### CODE REVIEW

File: src/map/battlefield_handler.cpp [hunk 1]
```diff
@@ -174,8 +174,18 @@ void CBattlefieldHandler::handleBattlefields(time_point tick){
...
 					else{
-						ShowDebug("BCNM %i battlefield %i : Winning conditions met. Exiting battlefield.\n",PBattlefield->getID(),PBattlefield->getBattlefieldNumber());
-						PBattlefield->winBcnm();
+						// BCNMs with no treasure chest ends around 7 seconds after winning.
+						if (!PBattlefield->won()){
+							ShowDebug("BCNM %i battlefield %i : Winning conditions met. Waiting 7 seconds before exiting battlefield.\n",
+									PBattlefield->getID(), PBattlefield->getBattlefieldNumber());
+							PBattlefield->win();
+						} else {
+							if (std::chrono::duration_cast<std::chrono::seconds>(tick - PBattlefield->getWinTime()).count() > 6){
```
Comment (teschnei) [inline src/map/battlefield_handler.cpp]: if (tick - PBattlefield->getWinTime() > 6s)

**Comment by teschnei** [comment]
what about bcnms where victory isnt a kill (ouryu, tenzen)?

**Comment by dickeyf** [comment]
Hmmm... I didnt think about those.  I can take a look tomorow (WHM level
cap too fits here I think)

**Comment by dickeyf** [comment]
Added a clause that ends the BCNM right away when victory condition is met but mobs didn't die.  Tested on Ouryu BCNM

File: scripts/zones/Throne_Room/Zone.lua [hunk 2]
```diff
@@ -63,4 +63,8 @@ end;
 function onEventFinish(player,csid,option)    
     --printf("CSID: %u",csid);
     --printf("RESULT: %u",option);
+    if (csid == 0x06) then
+        -- You will be transported back to the entrance of Castle Zvahl Baileys
+        player:setPos(378.222,-12,-20.299,125,0xA1);
```
Comment (teschnei) [inline scripts/zones/Throne_Room/Zone.lua]: the cutscene should move you without the position needing to be set manually

Comment (dickeyf) [inline scripts/zones/Throne_Room/Zone.lua]: I just tried and the CS moves to the entrance of Throne room.  It won't zone you at the entrance of Castle Zvahl Baileys.  I don't CS can zone you, only move you.  According to original script, we want to zone.

**Comment by dickeyf** [comment]
I just tried and the CS moves to the entrance of Throne room. It won't zone you at the entrance of Castle Zvahl Baileys. I don't CS can zone you, only move you. According to original script, we want to zone. I also just fixed the CS id, should have been 7,not 6.

**Pull request merged**